dashboard/app/controllers/pd/fit_weekend1819_registration_controller.rb

```diff
@@ -0,0 +1,35 @@
+class Pd::FitWeekend1819RegistrationController < ApplicationController
+  load_and_authorize_resource :application,
+    class: 'Pd::Application::ApplicationBase', find_by: :application_guid,
+    id_param: :application_guid
+
+  # here we handle the CanCan error manually so that we can present
+  # non-authorized users with a custom page explaining that they must be logged
+  # in as a facilitator account, rather than giving them the generic 404
+  rescue_from CanCan::AccessDenied do
+    render :unauthorized
+  end
+
+  def new
+    if Pd::FitWeekend1819Registration.exists?(pd_application_id: @application.id)
+      render :submitted
+      return
+    end
+
+    @script_data = {
+      props: {
+        options: Pd::FitWeekend1819Registration.options.camelize_keys,
+        requiredFields: Pd::FitWeekend1819Registration.camelize_required_fields,
+        apiEndpoint: "/api/v1/pd/fit_weekend_registrations",
+        applicationId: @application.id,
+        course: @application.fit_workshop.course,
+        city: @application.fit_workshop.location_city,
+        date: @application.fit_workshop.friendly_date_range,
+        email: @application.user.email,
```

Comment:
Should we use the email in the application hash rather than their user email? Just in case they have a new email but haven't updated their account?

Reply:
Good question! @tanyaparker, thoughts?

Reply:
(going to go ahead and merge this now and consider this change in a separate PR).

Reply:
Don't we use the user email for the application? As in, we don't allow them to submit with another email other than adding in a secondary email. So aren't these the same?

Reply:
you're right- I forgot how the application worked when I asked this.